.github/workflows/ci.yml

```diff
@@ -27,6 +28,12 @@ jobs:
         env:
           PYTHON: ""
           MANIFOLDS_TEST_GROUP: ${{ matrix.group }}
+        if: ${{ matrix.group == 'test_on_default_compiler' }}
+      - run: julia --color=yes --depwarn=yes --optimize=0 --compile=min --inline=yes --project=@. -e 'import Pkg;include(joinpath(ENV["GITHUB_ACTION_PATH"], "kwargs.jl"));kwargs = Kwargs.kwargs(;coverage = :(true),force_latest_compatible_version = :(auto), julia_args = ["--check-bounds=yes"]);Pkg.test(; kwargs...)
+        shell: bash
```

Comment:
After this I'm definitely convinced https://github.com/julia-actions/julia-runtest/issues/47 either needs fixing or julia-runtest needs a refactoring and better docs to make it easier to understand what kwargs is even doing.

As a quick workaround until then, the command you're running depends on [`kwargs.jl`](https://github.com/julia-actions/julia-runtest/blob/master/kwargs.jl) which depends on [`autodetect-dependabot.jl`](https://github.com/julia-actions/julia-runtest/blob/master/autodetect-dependabot.jl). My suggestion would be to copy those files into `.github/workflows/` until there's a proper fix (those files are MIT-licensed, so as long as you copy [`LICENSE`](https://github.com/julia-actions/julia-runtest/blob/master/LICENSE) to the top of the files or include it somewhere in the repo, e.g. in a `THIRD_PARTY_NOTICE.md`, copying the files is fine).

If you don't want to do that, an alternative would be to essentially copy their contents and strip out unnecessary stuff like the version check.

---

Unrelated to the above:

GHA let us define custom shells, which lets us treat everything within `run: |` as if it was a Julia file. `shell: julia --color=yes --depwarn=yes --optimize=0 --compile=min --inline=yes --project=@. {0}` creates a file with the content of the `run` block above it and replaces `{0}` with the path to that file. This isn't strictly needed but helps us declutter what's in `run` and hopefully make it easier to debug and follow the script.

```suggestion
      - run: |
          import Pkg
          include(joinpath(ENV["GITHUB_WORKSPACE"], ".github", "workflows", "kwargs.jl"))
          kwargs = Kwargs.kwargs(;coverage = :(true),force_latest_compatible_version = :(auto), julia_args = ["--check-bounds=yes"])
          Pkg.test(; kwargs...)
        shell: julia --color=yes --depwarn=yes --optimize=0 --compile=min --inline=yes --project=@. {0}
```

Reply:
Oh I could not get the kwargs to work and then noticed it is extra Julia code you have so I removed that for now, but it seems actually necessaray?

Reply:
Hm I have to really carefully read this, you seem to really know the stuff and I beryl understand what you ware writing, sorry.

Reply:
Maybe I'll just wait what happens to the action, because I fear I could also spend a few weeks on this otherwise.

Reply:
> Oh I could not get the kwargs to work and then noticed it is extra Julia code you have so I removed that for now, but it seems actually necessaray?

The main thing it does that is also necessary here is figuring out the value of `force_latest_compatible_version`. Its value is different if the PR has been opened by CompatHelper (or Dependabot if that ever receives Julia support). The motivation for that behaviour is outlined here: https://github.com/JuliaLang/Pkg.jl/pull/2439 -- However, you can also choose to ignore that for now and remove `force_latest_compatible_version ` or alternatively have two completely different `run` blocks for the normal case vs the dependabot case with an `if`.

> Maybe I'll just wait what happens to the action, because I fear I could also spend a few weeks on this otherwise.

That works as well, I can't promise anything but it's on my list to look at for this week.

Reply:
> Hm I have to really carefully read this, you seem to really know the stuff and I beryl understand what you ware writing, sorry.

No worries at all, that just means I have to explain/write things better

Reply:
Ah, thanks for the explanation :)

And concerning the update for runtest – do not hurry and all is fine :) I am very happy to have the runtests action – this PR (if it finally works) just would speedup our CI from 60 to 20 minutes (doing most of the tests with `-C0 --compile=min`), so the worst thing is – that our CI stays slow.
And – both this and the runtest action are open source things – so – really nothing of a hurry here. Thanks again for providing the action and for considering the addition of the arguments. And your help even here Really appreciated 👍